src/lib/media-query/match-media-provider.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {
+  NgZone,
+  PLATFORM_ID,
+  InjectionToken,    // tslint:disable-line:no-unused-variable
+} from '@angular/core';
+import {DOCUMENT, isPlatformServer} from '@angular/common';
+
+import {MatchMedia} from './match-media';
+import {ServerMatchMedia} from './server-match-media';
+
+/**
+ * Create a version of MatchMedia compatible with the current
+ * platform
+ */
+export function MEDIA_QUERY_LIST_FACTORY(platformId: Object,
```

Comment:
Instead of having one provider that returns either service, there should just be two different providers. The appropriate provider is then included in each of the browser / server module.

Reply:
Fixed